presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/PredicatePushDown.java

```diff
@@ -727,29 +727,45 @@ private static DynamicFiltersResult createDynamicFilters(
             return clausesBuilder.build();
         }
 
+        private static boolean isComparisonOperator(CallExpression call, JoinNode node)
+        {
+            if (call.getDisplayName().equals(EQUAL.getFunctionName().getObjectName()) ||
+                    call.getDisplayName().equals(LESS_THAN.getFunctionName().getObjectName()) ||
+                    call.getDisplayName().equals(LESS_THAN_OR_EQUAL.getFunctionName().getObjectName()) ||
+                    call.getDisplayName().equals(GREATER_THAN.getFunctionName().getObjectName()) ||
+                    call.getDisplayName().equals(GREATER_THAN_OR_EQUAL.getFunctionName().getObjectName())) {
+                List<RowExpression> arguments = call.getArguments();
+                RowExpression left = arguments.get(0);
+                RowExpression right = arguments.get(1);
+                Set<VariableReferenceExpression> leftUniqueOutputs = VariablesExtractor.extractUnique(left);
+                Set<VariableReferenceExpression> rightUniqueOutputs = VariablesExtractor.extractUnique(right);
+                return (node.getRight().getOutputVariables().containsAll(rightUniqueOutputs) && node.getLeft().getOutputVariables().containsAll(leftUniqueOutputs)) ||
+                    (node.getRight().getOutputVariables().containsAll(leftUniqueOutputs) && node.getLeft().getOutputVariables().containsAll(rightUniqueOutputs));
```

Comment:
This logic is not about whether the operator is a comparison operator. What does it do? Please name the function to reflect the logic. Thanks!

Reply:
yep, supported operator should be comparison, excluding not_equal and distinct, and the expression needs:
1. left child contains left variables and right child contains right variables
2. or left child contains right variables and right child contains left variables

get it updated